contentcuration/contentcuration/management/commands/exportchannel.py

```diff
@@ -245,76 +247,105 @@ def process_assessment_metadata(ccnode, kolibrinode):
-        }
-        exercise_result = render_to_string('perseus/exercise.json', exercise_context)
-        zf.writestr("exercise.json", exercise_result)
-
-        for question in ccnode.assessment_items.all():
-            for image in question.files.filter(preset_id=format_presets.EXERCISE_IMAGE):
-                image_name = "images/{0}.{ext}".format(image.checksum, ext=image.file_format_id)
-                if image_name not in zf.namelist():
-                    image.file_on_disk.open(mode="rb")
-                    zf.writestr(image_name, image.file_on_disk.read())
-
-            for image in question.files.filter(preset_id=format_presets.EXERCISE_GRAPHIE):
-                svg_name = "images/{0}.svg".format(image.original_filename)
-                json_name = "images/{0}-data.json".format(image.original_filename)
-                if svg_name not in zf.namelist() or json_name not in zf.namelist():
-                    image.file_on_disk.open(mode="rb")
-                    content=image.file_on_disk.read()
-                    content = content.split(exercises.GRAPHIE_DELIMITER)
-                    zf.writestr(svg_name, content[0])
-                    zf.writestr(json_name, content[1])
-
-        for item in ccnode.assessment_items.all():
-            write_assessment_item(item, zf)
+        try:
+            exercise_context = {
+                'exercise': json.dumps(exercise_data, sort_keys=True, indent=4)
+            }
+            exercise_result = render_to_string('perseus/exercise.json', exercise_context)
+            write_to_zipfile("exercise.json", exercise_result, zf)
+
+            for question in ccnode.assessment_items.all().order_by('order'):
+                for image in question.files.filter(preset_id=format_presets.EXERCISE_IMAGE).order_by('checksum'):
+                    image_name = "images/{0}.{ext}".format(image.checksum, ext=image.file_format_id)
+                    if image_name not in zf.namelist():
+                        image.file_on_disk.open(mode="rb")
+                        write_to_zipfile(image_name, image.file_on_disk.read(), zf)
+
+                for image in question.files.filter(preset_id=format_presets.EXERCISE_GRAPHIE).order_by('checksum'):
+                    svg_name = "images/{0}.svg".format(image.original_filename)
+                    json_name = "images/{0}-data.json".format(image.original_filename)
+                    if svg_name not in zf.namelist() or json_name not in zf.namelist():
+                        image.file_on_disk.open(mode="rb")
+                        content = image.file_on_disk.read()
+                        content = content.split(exercises.GRAPHIE_DELIMITER)
+                        write_to_zipfile(svg_name, content[0], zf)
+                        write_to_zipfile(json_name, content[1], zf)
+
+            for item in ccnode.assessment_items.all().order_by('order'):
+                write_assessment_item(item, zf)
+
+        finally:
+            zf.close()
```

Comment:
Isn't this `zf.close` guaranteed to be executed after the `with` statement already?